frontend/js/src/artist-similarity/src/artist-search/SearchBox.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useEffect, useMemo } from "react";
+import SearchDropdown from "./SearchDropdown";
+import ArtistLookup from "./ArtistLookup";
+import { ArtistType } from "./ArtistLookup";
+import { faMagnifyingGlass, faMinus, faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import "./SearchBox.css";
+interface SearchBoxProps {
+    currentsimilarArtistsLimit: number;
+    onsimilarArtistsLimitChange: (similarArtistsLimit: number) => void;
+    currentArtistMbid: string;
+    onArtistChange: (artist_mbid: string) => void;
+}
+
+const SearchBox = (props: SearchBoxProps) => {
+    // State to store the search results (list of artists)
+    const [searchResults, setSearchResults] = React.useState<Array<ArtistType>>([]);
+    const [searchQuery, setSearchQuery] = React.useState<string>("");
+    const [similarArtistsLimit, setsimilarArtistsLimit] = React.useState<number>(props.currentsimilarArtistsLimit);
+    const [artistMbid, setArtistMbid] = React.useState<string>(props.currentArtistMbid);
+    // Delay before dropdown disappears to make sure user can click on it
+    const dropdownDelay = 200;
+
+    // Lookup the artist based on the query
+    const getArtists = async (): Promise<void> => {
+        if(searchQuery !== "" || searchQuery.trim() !== ""){
+            const results = await ArtistLookup(searchQuery);
+            setSearchResults(results ?? []);
+        }
+        else{
+            setSearchResults([]);
+        }
+    }
+    // Lookup the artist based on the query
+    useEffect(() => {
+        getArtists();
+    }, [searchQuery]);
+    
+    // Set similarArtistsLimit based on user input
+    const handlesimilarArtistsLimitChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        let similarArtistsLimit = event.currentTarget.valueAsNumber;
+        setsimilarArtistsLimit(similarArtistsLimit);
+    }
+    // Set the artist and similarArtistsLimit based on user input
+    useEffect(() => {
+        props.onsimilarArtistsLimitChange(similarArtistsLimit);
+        props.onArtistChange(artistMbid);
+        setSearchQuery("");
+    }, [similarArtistsLimit, artistMbid]);
+
+    // Hide the dropdown when the user clicks outside of it
+    const toggleDropdown = () => setTimeout(() => {
+        const dropdown = document.getElementById("search-dropdown");
+        dropdown?.style.display === "flex" ? dropdown.style.display = "none" : dropdown!.style.display = "flex";
+    }, dropdownDelay);
```

Comment:
So I looked up the dropdown toggle on the add listens and created a similar setup. I use a state to handle it so that it can disappear when the user clicks on the same which it does not on the listens page.